### PR TITLE
FIX: Allow new client to work with js and ts next.js projects

### PR DESCRIPTION
**Pull request by logan-anderson**
If a `schema.js` is provided  then 
- `types.js`
- `types.d.s`
- `client.js`
is generated. 

If a `schema.{ts,tsx}` is provided then
- `types.ts`
- `client.ts`
is generated.

### CODE REVIEW

File: packages/@tinacms/cli/src/cmds/compile/index.ts [hunk 1]
```diff
@@ -36,15 +38,16 @@ export const resetGeneratedFolder = async ({
     console.log(e)
   }
   await fs.mkdir(tinaGeneratedPath)
+  const etx = usingTs ? 'ts' : 'js'
```
Comment (kldavis4) [inline packages/@tinacms/cli/src/cmds/compile/index.ts:41]: should this be ext?

File: .changeset/pretty-ties-lick.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@tinacms/cli': patch
+---
+
+Genearte types and client based on the file type of the schema
```
Comment (kldavis4) [inline .changeset/pretty-ties-lick.md:5]:
```suggestion
Generate types and client based on the file type of the schema
```

File: packages/@tinacms/cli/src/buildTina/index.ts [hunk 3]
```diff
@@ -167,6 +167,7 @@ export const build = async ({
   skipIndex,
 }: BuildOptions) => {
   const rootPath = ctx.rootPath as string
+  const tinaGeneratedPath = path.join(rootPath, '.tina', '__generated__')
```
Comment (kldavis4) [inline packages/@tinacms/cli/src/buildTina/index.ts:170]: should this be down below the rootPath null check?

File: packages/@tinacms/cli/src/cmds/compile/index.ts (hunk 1 above)
Comment (logan-anderson) [reply packages/@tinacms/cli/src/cmds/compile/index.ts]: yup, fixed.

File: packages/@tinacms/cli/src/buildTina/index.ts (hunk 3 above)
Comment (logan-anderson) [reply packages/@tinacms/cli/src/buildTina/index.ts]: Yes, fixed.

**Review by jamespohalloran — APPROVED** [review]
Tested this out in the barebones starter and it's working nicely!

**Pull request merged**